File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/MinItemsValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.List;

public class MinItemsValidator implements KeywordValidator {
    public final int minItems;

    public MinItemsValidator(int minItems) {
        this.minItems = minItems;
    }

    @Override
    public @Nullable PathToSchemasMap validate(
        ValidationData data
    ) {
        if (!(data.arg() instanceof List<?> listArg)) {
            return null;
        }
        if (listArg.size() < minItems) {
            throw new ValidationException("Value " + listArg + " is invalid because has < the minItems of " + minItems);
        }
        return null;
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/MinItemsValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.exceptions.ValidationException

class MinItemsValidator : KeywordValidator {
    @Throws(ValidationException::class)
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap? {
        val minItems: Int = data.schema.minItems ?: return null
        if (data.arg !is List<*>) {
            return null
        }
        if (data.arg.size < minItems) {
            throw ValidationException("Value ${data.arg} is invalid because has < the minItems of $minItems")
        }
        return null
    }
}
